cranelift/codegen/src/isa/x64/inst/mod.rs

```diff
@@ -1108,8 +1108,8 @@ impl PrettyPrint for Inst {
                 ..
             } => {
                 let src1 = pretty_print_reg(src1.to_reg(), 8, allocs);
-                let src2 = src2.pretty_print(8, allocs);
                 let dst = pretty_print_reg(dst.to_reg().to_reg(), 8, allocs);
+                let src2 = src2.pretty_print(8, allocs);
```

Comment:
Why does changing the order of this matter? The format string hasn't changed...

Reply:
`allocs` is an `AllocationConsumer`, so invocation order matters here; the order must match the order in which we provide the register arguments in `get_operands`. (Yep, it's pretty fragile, and I want to find a better way; this was a result of the "don't mutate instructions post-regalloc" design decision that was part of the RA2 speedup.)